sound/soc/sof/ipc4-pcm.c

```diff
@@ -302,17 +300,6 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 	if (!pipeline_list->pipelines || !pipeline_list->count)
 		return 0;
 
-	spipe = pipeline_list->pipelines[0];
-	pipe_widget = spipe->pipe_widget;
-	pipeline = pipe_widget->private;
-
-	/*
-	 * If use_chain_dma attribute is set we proceed to chained DMA
-	 * trigger function that handles the rest for the substream.
-	 */
-	if (pipeline->use_chain_dma)
-		return sof_ipc4_chain_dma_trigger(sdev, pipeline_list, state, cmd);
-
```

Comment:
It might be better to separate the second half of this function as a new one to let's say set_pipelines_state_worker or something?
It is starting to get odd to see triggers all around the code at callback levels which should not trigger at all.